Readme.md

```diff
@@ -8,11 +8,106 @@ Specifically, DSharp lets you leverage the experience of C# (intellisense, build
+
+    - **Mscorlib:** Dummy interfaces and classes that emulate the original `mscorlib.dll`. Having the same signatures of the original mscorlib enables cross-compilation.
+
+    - **Scripts:** Javascript implementations of mscorlib. It mostly includes helper methods and class implementations. The compiler binds C# code to JS implementation by using `ScriptAttributes`.
+
+    - **Web:** Web-specific abstractions that can be used when targeting DSharp. This is an extension library and isn't available in normal C#.
+
+    - **Shell:** Shell utility for invoking the compiler.
+
+- ### SDK
+
+    - Crosscompilation for DSharp and NetStandard
+    - Support for modern `csproj` syntax
+
+- ### Tests
+
+    - **Framework:** Implementation of the JS test framework.
+    - **Compiler.Tests:** Tests for the compiler.
+
+## Writing Tests ##
+
+Testing the compiler is quite straightforward: define an input c# file `Code.cs` and the expected result in `Baseline.txt`. The test runner will compare the result with the expected one.
+
+In order to create a new  you need to:
+
+1. Create the test files `Code.cs` and `Baseline.txt` in a relevant subdirectory under the `Source` folder.
+2. Add your test to `CompilerTests.json` pointing to the right files. The categories represent subdirectories inside the `Source` folder.
+
+## Debugging ##
+
+There are two ways of debugging the project: debugging a test or a shell that invokes the compiler.
+
+### Prerequisites ###
+
+- #### Enable all Common Language Runtime Exceptions
+  This can be done under `Debug` -> `Windows` -> `Exception Settings (Ctrl+Alt+E)` and ticking all `Common Language Runtime Exceptions`.
+
+### Debugging a Test ###
+
+Debugging a test is currently non-deterministic (it works depending on the computer, and it may display some artifacts on Visual Studio) but it's worth trying. This is due to xUnit runtime confusing Visual Studio.
+
+In order to debug a test you need to:
+
+1. Rebuild All
+2. In the test explorer, `Right Click` -> `Debug Selected Tests`
+
+### Debugging the Shell ###
+
+Debugging the Shell requires a bit of configuration: setting up the arguments for the invokation.
+
+In order to debug the shell you need to:
+
+1. Rebuild All
+2. In the `Core/DSharp.Shell` project, `Right Click` -> `Properties` -> `Debug` -> `Application arguments` and set the following:
+
+    ```
+    /ref:C:\Working\dsharp\src\DSharp.Mscorlib\bin\Debug\net461\DSharp.Mscorlib.dll
+    /out:out.js
+    C:\Working\dsharp\src\DSharp.Compiler.Tests\Source\Mscorlib\DateTime\Code.cs
+    ```
+    > update the paths to match the ones in your pc
+3. F5 in debug mode
+
+## Useful Links ##
+
+- ### mscorlib
+  Being compliant with latest implementations of `mscorlib` is extremely important for cross-compiling. These can be checked in the following links:
+
+  - [Source Dot Net](https://source.dot.net): Source reference with nice search and navigation
```

Comment:
Can we add Reference source here as well

Reply:
what is the difference exactly?

Reply:
Not quite, Reference source is the source of .net 4.8.1. And source.net is the source of the current DotNetCore -> 2.2.x?